src/javascript/components/datepicker.js

```diff
@@ -112,9 +107,10 @@ export default class Datepicker extends React.Component {
           ref="icon"
           type="button"
           className="FormFieldAddOn-item"
-          onClick={this.handleIconClick}
           tabIndex="-1"
-          dangerouslySetInnerHTML={{__html: iconCalendar}} />
+        >
+          <Icon type="event" />
```

Comment:
Does this work? I remember I did the dangerously set innerHTML thing for a reason, but maybe that reason no loner applies...

Reply:
So, the `<Icon>` component ALSO uses dangerouslySetInnerHTML, so this would work here either way. However, the reason you used dangerouslySetInnerHTML is that React didn't used to support the svg attribute "xlink:href". Support for it was added in [0.14](https://reactjs.org/blog/2015/10/07/react-v0.14.html#notable-enhancements), as "xlinkHref", so I updated the `<Icon>` component

Reply:
SGTM.

I'm mistaken, DOMParser is a good way to remove HTML from a string. What you have is fine. (BTW, the old was probably fine too. I found a jsperf showing browsers could do that like 800,000 times a second, so the perf hit is minimal).

The only gotcha I know of with the `_.escape` method is in cases where an attribute isn't quoted.